src/OrbitGl/BatchRenderGroup.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2023 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitGl/BatchRenderGroup.h"
+
+namespace orbit_gl {
+StencilConfig& StencilConfig::ClipAt(const StencilConfig& parent) {
+  if (!parent.enabled) return *this;
+  if (!enabled) {
+    *this = parent;
+    return *this;
+  }
+
+  pos[0] = std::min(std::max(pos[0], parent.pos[0]), parent.pos[0] + parent.size[0]);
+  pos[1] = std::min(std::max(pos[1], parent.pos[1]), parent.pos[1] + parent.size[1]);
+
+  size[0] = std::max(std::min(size[0], parent.pos[0] + parent.size[0] - pos[0]), 0.f);
+  size[1] = std::max(std::min(size[1], parent.pos[1] + parent.size[1] - pos[1]), 0.f);
```

Comment:
It would also be an option to use `QRect`  as a replacement for `size` and `pos`. It has all the operations that we need - including the intersection of two QRects. Not sure though if it makes sense to change that now. Just an option.

Reply:
As discussed - this is too much of a mess in the rest of the code. Changed this to now use Vec2.